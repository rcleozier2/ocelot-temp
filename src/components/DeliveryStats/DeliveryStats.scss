@import "../../App-Variables.scss";

.delivery-stats {
    background-color: #ffffff;
    padding: 10px;
    border: 1px solid #eee;
    text-align: center;
    justify-content: center;
    filter: drop-shadow(2px 2px 4px #dedede);
    min-height: 250px;
    height: auto;
    overflow: hidden;

    &-text {
        font-size: 13px;
        text-align: left;
    }
}