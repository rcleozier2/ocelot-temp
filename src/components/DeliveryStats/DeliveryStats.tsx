import React from "react";

import "./DeliveryStats.scss";

const DeliveryStats = (props: any) => {
  return (
    <>
      <div className="delivery-stats">
        <div className="row">
          <div className="col-12">
            <p className="delivery-stats-text">
              Completed On Time ({props.tasks.total.completedOntime})
            </p>
            <p className="delivery-stats-text">
              Completed Late ({props.tasks.total.completedLate})
            </p>
            <p className="delivery-stats-text">
              Total Completed ({props.tasks.total.completed})
            </p>
            <hr />
            <p className="delivery-stats-text">
              Failed On Time ({props.tasks.total.failedOntime})
            </p>
            <p className="delivery-stats-text">
              Failed Late ({props.tasks.total.failedLate})
            </p>

            <p className="delivery-stats-text">
              Total Failed ({props.tasks.total.failed})
            </p>
            <hr />

            <p className="delivery-stats-text">
              Total Scheduled ({props.tasks.total.scheduled})
            </p>
          </div>
        </div>
      </div>
    </>
  );
};

export default DeliveryStats;
