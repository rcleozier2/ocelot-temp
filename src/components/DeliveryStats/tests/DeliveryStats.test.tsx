import React from "react";
import DeliveryStats from "../DeliveryStats";
import renderer from "react-test-renderer";

it("renders DeliveryStats correctly", () => {
  const props = {
    tasks: {
      total: {
        completedOntime: 2,
        completedLate: 3,
        completed: 5,
        failedOntime: 2,
        failedLate: 2,
        failed: 4,
        scheduled: 9,
        justSends: 4
      }
    }
  };

  const component = renderer.create(<DeliveryStats {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders DeliveryStats correctly with new props", () => {
  const props = {
    tasks: {
      total: {
        completedOntime: 12,
        completedLate: 13,
        completed: 15,
        failedOntime: 12,
        failedLate: 12,
        failed: 14,
        scheduled: 29,
        justSends: 4
      }
    }
  };
  const component = renderer.create(<DeliveryStats {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
