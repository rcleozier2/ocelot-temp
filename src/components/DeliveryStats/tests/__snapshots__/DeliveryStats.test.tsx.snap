// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DeliveryStats correctly 1`] = `
<div
  className="delivery-stats"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p
        className="delivery-stats-text"
      >
        Completed On Time (
        2
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Completed Late (
        3
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Total Completed (
        5
        )
      </p>
      <hr />
      <p
        className="delivery-stats-text"
      >
        Failed On Time (
        2
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Failed Late (
        2
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Total Failed (
        4
        )
      </p>
      <hr />
      <p
        className="delivery-stats-text"
      >
        Just Sends (
        4
        )
      </p>
    </div>
  </div>
</div>
`;

exports[`renders DeliveryStats correctly with new props 1`] = `
<div
  className="delivery-stats"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p
        className="delivery-stats-text"
      >
        Completed On Time (
        12
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Completed Late (
        13
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Total Completed (
        15
        )
      </p>
      <hr />
      <p
        className="delivery-stats-text"
      >
        Failed On Time (
        12
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Failed Late (
        12
        )
      </p>
      <p
        className="delivery-stats-text"
      >
        Total Failed (
        14
        )
      </p>
      <hr />
      <p
        className="delivery-stats-text"
      >
        Just Sends (
        4
        )
      </p>
    </div>
  </div>
</div>
`;
